test/attribute_cache_test.rb

```diff
@@ -39,13 +39,13 @@ def test_nil_is_stored_in_the_cache_on_cache_misses
     end
   end
 
-  def test_cached_attribute_values_are_expired_from_the_cache_when_an_existing_record_is_saved
+  def test_cached_attribute_values_are_not_expired_from_the_cache_when_an_existing_record_is_saved_with_no_changes
```

Comment:
This is changing the purpose of this test.  Unless the test was redundant, we should be adding a new test for the new behaviour and just update this one to test the behaviour it was intended for.  This can be done by updating some attribute instead of calling `save!` without changing any attributes.